app/src/renderer/components/governance/ModalVote.vue

```diff
@@ -0,0 +1,176 @@
+<template lang="pug">
+  .modal-vote#modal-vote(v-click-outside="close")
+    .modal-vote-header
+      img.icon(class='modal-vote-atom' src="~assets/images/cosmos-logo.png")
+      span.tm-modal-title Vote
+      .tm-modal-icon.tm-modal-close#closeBtn(@click="close()")
+        i.material-icons close
+
+    div
+      h2 Title: {{ proposalTitle }}
+      h3 Proposal ID: {{ `#` + proposalId }}
+
+    tm-form-group.modal-vote-form-group.options
+      tm-btn#vote-yes(
+        @click.native="voteYes"
+        :class="[option ===  `yes` ? 'active' : '']"
+        color="secondary"
+        value="Yes"
+        size="md")
+
+      tm-btn#vote-no(
+        @click.native="voteNo"
+        :class="[option === `no` ? 'active' : '']"
+        color="secondary"
+        value="No"
+        size="md")
+
+      tm-btn#vote-veto(
+        @click.native="voteVeto"
+        :class="[option === `no_with_veto` ? 'active' : '']"
+        color="secondary"
+        value="No With Veto"
+        size="md")
+
+      tm-btn#vote-abstain(
+        @click.native="voteAbstain"
+        :class="[option === `abstain` ? 'active' : '']"
+        color="secondary"
+        value="Abstain"
+        size="md")
+
+    tm-form-group.modal-vote-form-group
+    .modal-vote-footer
+      tm-btn#cast-vote(
+        @click.native="onVote"
+        :disabled="$v.option.$invalid"
+        color="primary"
+        value="Vote"
+        size="lg")
+</template>
+
+<script>
+import ClickOutside from "vue-click-outside"
+import { required } from "vuelidate/lib/validators"
+import Modal from "common/TmModal"
+import { TmBtn, TmField, TmFormGroup, TmFormMsg } from "@tendermint/ui"
+
+const isValid = option =>
+  option === `yes` ||
+  option === `no` ||
+  option === `no_with_veto` ||
+  option === `abstain`
+
+export default {
+  name: `modal-vote`,
+  props: [`proposalId`, `proposalTitle`],
+  components: {
+    Modal,
+    TmBtn,
+    TmField,
+    TmFormGroup,
+    TmFormMsg
+  },
+  data: () => ({
+    option: ``
+  }),
+  validations() {
+    return {
+      option: {
+        required,
+        isValid
```

Comment:
In which case would this be invalid?

Reply:
correct me if I am wrong, but the user is only able to produce the states `yes` , `no`, `no_with_veto`, `abstain` and ``. so `isValid` equals `required`.

Reply:
correct, the button to vote is only disabled if in this case the user hasn't selected an option

Reply:
I just added it as a doublecheck ;)